src/dotnet/commands/dotnet-test/Program.cs

```diff
@@ -42,15 +42,35 @@ public int DoRun(string[] args)
                     RegisterForParentProcessExit(dotnetTestParams.ParentProcessId.Value);
                 }
 
-                var projectContexts = CreateProjectContexts(dotnetTestParams.ProjectPath);
-
-                var projectContext = projectContexts.First();
-
-                var testRunner = projectContext.ProjectFile.TestRunner;
+                var projectContexts = CreateProjectContexts(dotnetTestParams.ProjectPath, dotnetTestParams.Runtime);
```

Comment:
I believe you should break this part of the method in two: one for when a framework is passed in the params (and you will run only for that framework) and one for when you will be running for all frameworks.

You can have a conditional on dotnetTestParams.Framework != null and then call a RunTestsForFramework(). Otherwise call a RunTestsForAllFrameworks(). It simplifies this logic considerable and I believe you will even be able to remove this nasty ranTests.